tests/scripts/thread-cert/Cert_5_3_03_AddressQuery.py

```diff
@@ -31,121 +31,132 @@
-                time.sleep(1)
-
-        addrs = self.nodes[ROUTER3].get_addrs()
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertTrue(self.nodes[ED2].ping(addr))
-                time.sleep(1)
-
-        addrs = self.nodes[ROUTER3].get_addrs()
+        self.assertEqual(self.nodes[ROUTER1].get_state(), 'router')
+        self.assertEqual(self.nodes[DUT_ROUTER2].get_state(), 'router')
+        self.assertEqual(self.nodes[ROUTER3].get_state(), 'router')
+        self.assertEqual(self.nodes[MED1].get_state(), 'child')
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+
+        # 2
+        router3_mleid = self.nodes[ROUTER3].get_ip6_address(config.ADDRESS_TYPE.ML_EID)
+        self.assertTrue(self.nodes[MED1].ping(router3_mleid))
+
+        # Verify DUT_ROUTER2 sent a Address Query Request to the Realm local address.
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_messages.next_coap_message('0.02', '/a/aq')
+        command.check_address_query(msg, config.REALM_LOCAL_ALL_ROUTERS_ADDRESS) 
+
+        # 3
+        med1_mleid = self.nodes[MED1].get_ip6_address(config.ADDRESS_TYPE.ML_EID)
+        self.assertTrue(self.nodes[ROUTER1].ping(med1_mleid))
+
+        # Verify DUT_ROUTER2 respond with a Address Notification.
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_messages.next_coap_message('0.02', '/a/an')
+        command.check_address_notification(msg, self.nodes[DUT_ROUTER2], self.nodes[ROUTER1]) 
+
+        # 4
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        self.assertTrue(self.nodes[MED1].ping(router3_mleid))
+
+        # Verify DUT_ROUTER2 didn't send a Address Query Request. 
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_messages.next_coap_message('0.02', '/a/aq', False)
+        assert msg is None, "The Address Query Request is not expected."
+
+        # 5
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
         self.nodes[ROUTER3].stop()
-        time.sleep(140)
-
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertFalse(self.nodes[ED2].ping(addr))
-
-        addrs = self.nodes[ED2].get_addrs()
-        self.nodes[ED2].stop()
-        time.sleep(10)
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertFalse(self.nodes[BR].ping(addr))
+        time.sleep(550)
+        self.assertFalse(self.nodes[MED1].ping(router3_mleid))
+
+        # Verify DUT_ROUTER2 sent a Address Query Request to the Realm local address.
+        dut_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_messages.next_coap_message('0.02', '/a/aq')
+        command.check_address_query(msg, config.REALM_LOCAL_ALL_ROUTERS_ADDRESS) 
+
+        # 6
+        self.nodes[MED1].stop()
+        time.sleep(MED1_TIMEOUT)
```

Comment:
Test Plan (Step 6) expects two ping actions, one is to clear the cache and the other is to produce `a/aq`